### DISCUSSION TITLE
Invalid Flexlm Script for Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by tsibbett** [polarfire-soc/polarfire-soc-documentation#190] [General] [answered]
The `linux_setup_libero_soc_ug.pdf` file defines a flexlm script that starts at boot. I corrected some of the syntax errors, e.g., 

1. removed the trailing in the first line of  #!/bin/bash # as this is an invalid shebang line
2. updated paths
3. deleted line 9  "running" as this was meant to be commented out
4. changed line ". /etc/init.d/functions # Check license file" to ". /lib/lsb/init-functions # Check license file" as the original path doesn't exist in Ubuntu
5. Moved '}' on line 22 to below the return $RETVAL
 
But still haven't gotten it to work due to some misaligned ')' characters. Any ideas on what else I need to do to have the license launching at startup?

**Comment by AntonKrug** [comment, marked as answer]
Not sure if the is the right forum to troubleshoot Libero issues here

**Reply by tsibbett** [reply]
Do you have a recommendation on where I should move it? Are there more general microsemi github repos?

**Reply by hughbreslin** [reply]
Hi @tsibbett the main place for tools issues is our [Tech Support](https://microchipsupport.force.com/s/) 🙂 Tech support do keep an eye here but we're trying to keep this forum focused on the sources we release on GitHub. I filed a bug report on this issue for you so its hopefully fixed for the next release of the doc.

**Comment by TommyMurphyTM1234** [comment]
Part of the problem is that the formatting of the script in the documentation is a mess (in particular, missing line breaks causing lines to be incorrectly concatenated, missing indentation etc.) so that if you copy and paste it from the document it will almost certainly not work without reformatting, and maybe other changes. For example:

![Screenshot_20220824-222152](https://user-images.githubusercontent.com/14550604/186526877-9b64753e-387c-4419-9afd-9aa638a072ba.png)

Just to be clear, I'm referring to the document titled _"How to Set Up Your Linux Environment for Libero SoC v12.0"_ accessible using this URL:

* https://www.microsemi.com/document-portal/doc_download/1243979-how-to-set-up-your-linux-environment-for-libero-soc-v12-0-or-later

from this page:

* https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/fpga/libero-software-later-versions

Also here:

* https://onlinedocs.microchip.com/pr/GUID-FABC58FF-E2CC-4557-BA80-9C03AAFAA2D2-en-US-5/index.html?GUID-C4C81F3E-A5EC-499F-B6C0-21C283B346BA

The various references to different Libero versions (v12.0,   v2021.2, v2022.2) are another unnecessary cause for confusion, but, presumably, the information in this document is supposed to also apply to the latest Libero SoC release.

If you're running Libero SoC and the license daemon co-located on the same host then it might be simpler to take the sort of approach outlined in this link: 

* https://help.ubuntu.com/community/LiberoSOC#My_way_of_running_the_program

In this case a simple script is used to run the license daemon "on demand" before running Libero, thus obviating the need for running the former as a "service" daemon. And the license daemon is terminated when you're finished with Libero.

That also means that the license daemon isn't hogging resources when not needed. However it may mean that Libero SoC startup is slightly slower because the license daemon has to be started first. 

However this approach is no use if you need to configure the license daemon on a shared server accessed remotely by other hosts.

**Reply by tsibbett** [reply]
Thanks for the response. I agree with you entirely with regards to improper formatting - maybe it was correct at one time but the script provided on that PDF (accessed by the URLs mentioned above) is definitely broken without manual rework. The script that you provided at the end is also a good temporary solution and will work for now. I appreciate the help!

**Reply by AntonKrug** [reply]
I have dejavu felling. I think I was opening bug reports for exactly these mistakes for the Libero documentation back in 2017 :)
